src/Module.Server/Modes/Battle/CrpgBattleServer.cs

```diff
@@ -123,15 +136,15 @@ public override bool CheckForRoundEnd()
             return false;
         }
 
-        if (_flags.Length != 0 && Math.Abs(_morale) >= 1.0)
+        if (!_flagSystem.HasNoFlags() && Math.Abs(_morale) >= 1.0)
         {
-            if (!_wereFlagsRemoved)
+            if (!_flagSystem.WereFlagsManipulated())
             {
                 return true;
             }

```

Comment:
Check can be removed, because 
Math.Abs(_morale) >= 1.0 && !_flagSystem.WereFlagsManipulated() is always equal to false